crates/rattler/src/install/mod.rs

```diff
@@ -587,6 +614,9 @@ pub fn link_package_sync(
     // Ensure target directory exists
     fs_err::create_dir_all(target_dir).map_err(InstallError::FailedToCreateTargetDirectory)?;
 
+    #[cfg(target_os = "macos")]
+    exclude_from_backups(Path::new(&target_prefix));
```

Comment:
hmm, this function will be called for each package that is going to be linked (which might be 100+ times) while we only need to call it once.

I think the ideal solution would be to port over the `Prefix` type from `pixi` that would take care of all of these initializations upon calling `Prefix::create(opts)`.

All functions that curerntly take a `target_prefix: &Path` shoudl then take a `target_prefix: &Prefix` and `Prefix` should implemnet something like `AsRef<&Path>` or have a `prefix.path()` function.

I would also be OK with merging this without the `Prefix` type refactor, but then I think we should call this function from another place where it is only called once when linking packages.